### PR TITLE
colpack: refactor builder and build on darwin

### PR DESCRIPTION
**Pull request by smancill**
###### Motivation for this change

Need it to build on Darwin to build `shogun` on Darwin.

###### Things done

- Use `autoreconfHook` and set flags with `configureFlags`.
- Only enable OpenMP on Linux, so it can be built on Darwin.
- Do not install `.la` file. Other distros also remove them.
- Do not build examples.
- Install example sources into a proper location (or maybe they can just not be installed at all?)

This should make `shogun` build on Darwin but it probably won't build  due to being pinned to `gcc8`. I am working on `shogun` to make it work on Darwin so I will sent a new PR soon.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change (only `shogun`).
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134390` at 258ef732 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>colpack</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl3` is a deprecated license, please check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/science/math/colpack/default.nix:39:5:
```
   |
39 |     license = licenses.lgpl3;
   |     ^
```
</li>
  </ul>
</details>

**Comment by smancill** [comment]
Fixed license as suggested by the bot.

https://github.com/CSCsw/ColPack/blob/v1.0.10/Main/Definitions.h#L7-L10

File: pkgs/applications/science/math/colpack/default.nix [hunk 1]
```diff
@@ -12,20 +12,32 @@ stdenv.mkDerivation rec {
...
+    "--enable-openmp=${if stdenv.isLinux then "yes" else "no"}"
+    "--enable-examples=no"
+  ];
+
+  postInstall = ''
+    # Remove libtool archive
+    rm $out/lib/*.la
+
+    # Remove compiled examples (Basic examples get compiled anyway)
+    rm -rf $out/examples
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/colpack/default.nix:27]:
```suggestion
    rm -r $out/examples
```

Comment (smancill) [reply pkgs/applications/science/math/colpack/default.nix]: Fixed in latest force-push.

**Pull request merged**